mlflow/models/model.py

```diff
@@ -996,6 +996,12 @@ def update_model_requirements(
     old_conda_reqs = _get_requirements_from_file(conda_yaml_path)
     old_requirements_reqs = _get_requirements_from_file(requirements_txt_path)
 
+    try:
+        requirement_list = [Requirement(s.strip().lower()) for s in requirement_list]
+    except Exception as e:
+        raise MlflowException.invalid_parameter_value(
+            f"Invalid requirement list: {requirement_list}"
+        ) from e
```

Comment:
/ai make two changes

1. Replace `Exception` with `InvalidRequirement`
2. Only inclue invalid requirements in the error message.